jobs/elasticsearch/monit

```diff
@@ -1,8 +1,19 @@
+<% if p("elasticsearch.monitor.type") == "process") %>
 check process elasticsearch
   with pidfile /var/vcap/sys/run/elasticsearch/elasticsearch.pid
   start program "/var/vcap/jobs/elasticsearch/bin/monit_debugger elasticsearch_ctl '/var/vcap/jobs/elasticsearch/bin/elasticsearch_ctl start'" with timeout 120 seconds
   stop program "/var/vcap/jobs/elasticsearch/bin/monit_debugger elasticsearch_ctl '/var/vcap/jobs/elasticsearch/bin/elasticsearch_ctl stop'"
   group vcap
+<% else if p("elasticsearch.monitor.type") == "host" %>
+check host elasticsearch with address 127.0.0.1
+  start program "/var/vcap/jobs/elasticsearch/bin/monit_debugger elasticsearch_ctl '/var/vcap/jobs/elasticsearch/bin/elasticsearch_ctl start'" with timeout 120 seconds
+  stop program "/var/vcap/jobs/elasticsearch/bin/monit_debugger elasticsearch_ctl '/var/vcap/jobs/elasticsearch/bin/elasticsearch_ctl stop'"
+  if failed url http://127.0.0.1:9200/_cluster/health
```

Comment:
WDYT about adding a `and content == '"status":"green"'` check here so this will block until the cluster returns to a good state?

It'll slow down the deployment, but prevent the cluster from going red in this scenario:
- Date node 1 goes down for an update
- This causes it's primary shards to be marked unavailable (and the backup shards are activated).  
- The node finishes updating, ES restarts and rejoins the cluster (which is now in a yellow state as new backups shards are still being allocated) and monit marks the node up.
- Data node 2 goes down for the update
- This causes it's primary shards to be marked unavailable (and the backup shards are activated).
- If any of  node 2's primary shards were newly promoted backup shards from node 1 going down that haven't finished allocating backup shards yet, then the cluster is now red.